Sphinx Docs for Vector module

srjoglekar246 opened the pull request:
Adds Sphinx docs for the new sympy.vector module

srjoglekar246 committed:
Added files from physics.vector docs - unchanged

srjoglekar246 committed:
Added new file

srjoglekar246 committed:
Started with new file

srjoglekar246 committed:
Started working on fields file

srjoglekar246 committed:
Merge branch 'master' of https://github.com/sympy/sympy into vectordoc

srjoglekar246 committed:
docs v1

srjoglekar246 committed:
Merge branch 'master' of https://github.com/sympy/sympy into vectordoc

srjoglekar246 committed:
Fixing reviews- 1 [ci skip]

srjoglekar246 committed:
Merge branch 'master' of https://github.com/sympy/sympy into vectordoc

srjoglekar246 committed:
Fixed doctest errors

srjoglekar246 committed:
Created new intro file [ci skip]

srjoglekar246 committed:
Added examples

srjoglekar246 committed:
Merge branch 'master' of https://github.com/sympy/sympy into vectordoc

srjoglekar246 committed:
Merge branch 'master' of https://github.com/sympy/sympy into vectordoc

srjoglekar246 commented on doc/src/modules/vector/basics.rst:
```diff
@@ -0,0 +1,254 @@
+be initialized in :mod:`sympy.vector` as
+
+  >>> from sympy.vector import CoordSysCartesian
+  >>> N = CoordSysCartesian('N')
```
I just thought the names were getting a little too long. And yeah, that was the idea behind calling it the 'rectangular' system.
So should I go with `RectangularCoordinateSystem'?

moorepants replied:
If this is already in the 0.7.6 release we shouldn't change it.

srjoglekar246 commented on doc/src/modules/vector/basics.rst:
```diff
@@ -0,0 +1,254 @@
+  <class 'sympy.vector.vector.VectorZero'>
+  >>> N.i + Vector.zero
+  N.i
+  >>> Vector.zero == 2*Vector.zero
```
This was to keep consistency with SymPy's `S.Zero`

srjoglekar246 commented on doc/src/modules/vector/basics.rst:
```diff
@@ -0,0 +1,254 @@
+  >>> v1.dot(v2)
+  -11
+  >>> v1.cross(v2)
+  (-1)*N.i + (-3)*N.j + (-11)*N.k
```
I kept it this way so that the distinction between the coefficient and the basis vector parts is clear from looking at the expression itself. The bracketed expressions are always the measure numbers.

srjoglekar246 commented on doc/src/modules/vector/basics.rst:
```diff
@@ -0,0 +1,254 @@
+  >>> from sympy import Integral
+  >>> v1 = a*N.i + sin(a)*N.j - N.k
+  >>> Integral(v1, a)
+  (Integral(a, a))*N.i + (Integral(sin(a), a))*N.j + (Integral(-1, a))*N.k
```
As I mentioned before, I have tried to make it easy to understand the measure numbers and basis vectors involved in a vector expression by using brackets. This 'auto-evaluation' was put in for two reasons-
1. The way in which a vectorial expression gets integrated by a scalar is satisfied explicitely by this format.
2. It was causing some errors if kept in an Integral instance(like VectorMuls being converted into Muls), since that class does not deal with vectors. Therefore, by making it a vector instance immediately, these problems are avoided too.

srjoglekar246 commented on doc/src/modules/vector/basics.rst:
```diff
@@ -0,0 +1,254 @@
+in the form of a tuple. This is done using the ``express_coordinates``
+method.
+
+  >>> Q.express_coordinates(N)
```
+1 to this.

srjoglekar246 commented on doc/src/modules/vector/coordsys.rst:
```diff
@@ -0,0 +1,273 @@
+  >>> from sympy.vector import BodyOrienter
+  >>> from sympy.abc import a, b, c
+  >>> body_orienter = BodyOrienter(a, b, c, 'XYZ')
+  >>> M = N.orient_new('M', (axis_orienter, body_orienter))
```
Thanks!

srjoglekar246 commented on doc/src/modules/vector/coordsys.rst:
```diff
@@ -0,0 +1,273 @@
+  >>> M.position_wrt(N)
+  2*N.j
+  >>> from sympy.vector import express
+  >>> express(N.position_wrt(M), M)
```
Will add this. However, using the express function is more general, since it can re-express any quantity (scalars/vectors/dyadics) in a given system.

srjoglekar246 commented on doc/src/modules/vector/intro.rst:
```diff
@@ -0,0 +1,109 @@
+a field may also be dependent on time and other custom variables). Since we 
+only deal with 3D spaces in this module, a field is defined as a function of 
+the :math:`x`, :math:`y` and :math:`z` coordinates corresponding
+to a location in 3D space.
```
Thats a better way of putting it, yes. Will make the change.

jcrist commented:
@srjoglekar246 Any chance you can finish this up?

srjoglekar246 committed:
Merge branch 'master' of https://github.com/sympy/sympy into vectordoc

srjoglekar246 commented on doc/src/modules/vector/intro.rst:
```diff
@@ -0,0 +1,109 @@
+the position vector of one system's origin with respect to the other.
+
+Fields
+======
```
Adding a line clarifying this.

srjoglekar246 commented on doc/src/modules/vector/intro.rst:
```diff
@@ -0,0 +1,109 @@
+where , as per standard convention, :math:`\mathbf{\hat{i}}`,
+:math:`\mathbf{\hat{j}}` and :math:`\mathbf{\hat{k}}` represent unit vectors
+in the :math:`\mathbf{X}`, :math:`\mathbf{Y}` and :math:`\mathbf{Z}`
+directions respectively. Therefore, it can be concluded that the distance
```
+1.

srjoglekar246 committed:
Merge branch 'master' of https://github.com/sympy/sympy into vectordoc

srjoglekar246 committed:
Made minor wording changes

srjoglekar246 commented:
Okay, @moorepants and @jcrist 
I will finish up work on the first iteration of the vector module in two PRs. This one, and one other.
This PR will have following changes in the coming commits-
a. More diagrams in the explanation about coordinate systems. Do you know any efficient ways I could generate such vector diagrams?
b. I will stress that overloaded operators are not the recommended way to do things with the objects.
c. 'express_coordinates' will be changed to 'coordinates_in'.
All the other things, I have commented on, or rectified.

The other PR will have-
a. 'express' as a method on all BasisDependent classes.
b. Rename BaseDyadic to Dyad.
c. Set an appropriate value for is_commutative and other such attributes for all BasisDependent subclasses.

moorepants commented:
> More diagrams in the explanation about coordinate systems. Do you know any efficient ways I could generate such vector diagrams?

I use Inkscape or xfig. You can even use OpenOffice draw I think (not sure if it exports svg).

>  c. 'express_coordinates' will be changed to 'coordinates_in'.

If the vector module is 0.7.6 I don't think there is any reason to do this.

srjoglekar246 commented on doc/src/modules/vector/api/vectorfunctions.rst:
```diff
@@ -0,0 +1,55 @@
+================================================
```
I will add this in the next PR.

srjoglekar246 committed:
Added orientation diagram for coordsys file, changed examples

srjoglekar246 commented:
Added a diagram to explain the different orientation APIs over a common example.
Added a note mentioning that using overloaded operators is not the best way to perform vector operations, to maintain clarity of code.

srjoglekar246 committed:
Fixed error in docs

srjoglekar246 commented:
@jcrist Any idea why the Travis build is failing?

srjoglekar246 committed:
Merge branch 'master' of https://github.com/sympy/sympy into vectordoc

srjoglekar246 committed:
Named image properly in doc file

srjoglekar246 commented:
I am at a loss here. @skirpichev , can you take a look at the build error?

moorepants commented:
The latex warnings are pushing the travis log over 10000 lines and the error isn't present in the first lines. I'm not sure what all that is about. I'll try to build this PR locally.

toolforger commented:
The warnings are normal, they show up in successful builds as well.
I downloaded the log, the last lines are:

```
[1525] [1526] [1527] [1528]
Overfull \vbox (1.3029pt too high) detected at line 126890
[1529] [1530]
Overfull \vbox (0.8799pt too high) detected at line 127033
[1531] [1532] [1533]
! Unable to load picture or PDF file 'coordsys_rot.svg'.
<to be read again> 
                   }
l.127195 {\hfill\includegraphics{coordsys_rot.svg}
                                                  \hfill}
? 

No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.

The build has been terminated
```

So it's just a stalled build.

srjoglekar246 commented:
If thats the case, can we merge this once you have reviewed it @moorepants ?
I have opened a new PR to add the new global functions and methods.

moorepants commented:
The docs build on my machine.

srjoglekar246 commented:
Should I merge this one then?

moorepants commented:
I think there is one issue. See the docs makefile, line 12:

```
SVGFILES = $(wildcard src/modules/physics/vector/*.svg) $(wildcard src/modules/physics/mechanics/examples/*.svg)
```

I think you need to add the wildcard for your module.

moorepants commented:
After that I'm fine merging.

srjoglekar246 committed:
Added wildcard for vector module

srjoglekar246 commented:
Can this be merged?

moorepants commented:
Hmm, if we merge it and everyone starts getting this failure on the sphinx build, that may be an issue. I'm thinking we need to figure out why is is timing out.

moorepants commented:
I restarted the builds to try it one more time.

toolforger commented:
I've been seeing this and similar timeouts for quite a while now; they're seemingly random.
I suspect that the install script runs into some timeout and doesn't talk about it.

moorepants commented:
It is timing out in the sphinx build it seems. I don't see any evidence of timing out in the install.

moorepants commented:
This is odd:

```
+make man

Makefile:121: warning: overriding commands for target `_build/cheatsheet/cheatsheet.pdf'

Makefile:116: warning: ignoring old commands for target `_build/cheatsheet/cheatsheet.pdf'

docbook2x-man --to-stdout man/isympy.xml > man/isympy.1

I/O error : Attempt to load network entity http://www.oasis-open.org/docbook/xml/4.4/docbookx.dtd

man/isympy.xml:42: warning: failed to load external entity "http://www.oasis-open.org/docbook/xml/4.4/docbookx.dtd"

]>

  ^
```

Doesn't seem to raise an error though.

moorepants commented:
I also do not get all of those LaTeX warnings when I build locally. I wonder if I have something installed that isn't being installed on Travis.

srjoglekar246 commented:
I dont get them either! The docs build fine on my machine.

moorepants commented:
Do you get the LaTeX warnings locally?

srjoglekar246 commented:
No. I tried building it twice.

moorepants commented:
I'm going to ask on the mailing list.

moorepants commented:
This one passes even though all the warnings are there: https://travis-ci.org/sympy/sympy/jobs/64998102

moorepants commented:
I wonder if `--no-install-recommends` in the `.travis.yml` cause the warnings.

```
 sudo apt-get install --no-install-recommends graphviz inkscape texlive texlive-xetex texlive-fonts-recommended texlive-latex-extra lmodern librsvg2-bin imagemagick docbook2x;
```
Could be worth trying without it, but the downloads/installs will likely take longer.

moorepants commented:
@srjoglekar246 There is an issue with this pr, see the log

https://s3.amazonaws.com/archive.travis-ci.org/jobs/65544480/log.txt

```
! Unable to load picture or PDF file 'coordsys_rot.svg'.
<to be read again> 
                   }
l.127195 {\hfill\includegraphics{coordsys_rot.svg}
                                                  \hfill}
? 

No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.
```

SVGs are not allowed in the LaTeX. You need to investigate how the other svgs get loaded properly.

moorepants commented:
This is exactly what @toolforger already posted.

moorepants commented:
This is probably what you are missing:

```
.. image:: vec_rep.*
   :height: 175
   :width: 350
   :align: center
```

You need a wildcard in the sphinx docs.

moorepants commented on doc/src/modules/vector/coordsys.rst:
```diff
@@ -0,0 +1,280 @@
+
+The orientation is shown in the diagram below:
+
+.. image:: coordsys_rot.svg
```
Here you need a wildcard instead of `svg`.

srjoglekar246 committed:
Changed svg to wildcard

srjoglekar246 committed:
Merge branch 'master' of https://github.com/sympy/sympy into vectordoc

srjoglekar246 commented:
Made the change now.

toolforger commented:
If you compare with a non-failing build, you'll see that these warnings are all what Travis does.
I guess it's still a good idea to `apt-get install` enough that these warnings go away.

toolforger commented:
BTW `apt-get install` should be fast, Travis uses local caching for the packages any build requests.

certik commented:
This looks good, thanks!
